packages/exchange/src/exchange/message.py

```diff
@@ -25,9 +25,21 @@ def validate_role_and_content(instance: "Message", *_: any) -> None:  # noqa: AN
             raise ValueError("Assistant message does not support ToolResult")
 
 
-def content_converter(contents: list[dict[str, any]]) -> list[Content]:
-    return [(CONTENT_TYPES[c.pop("type")](**c) if c.__class__ not in CONTENT_TYPES.values() else c) for c in contents]
-
+def content_converter(contents: list) -> list[Content]:                                                                                                                      
+     result = []                                                                                                                                                              
+     for c in contents:                                                                                                                                                       
+         if isinstance(c, dict) and 'type' in c:  # Structured content logic                                                                                                  
+             content_type = c.pop('type')                                                                                                                                     
+             if content_type in CONTENT_TYPES:                                                                                                                                
+                 result.append(CONTENT_TYPES[content_type](**c))                                                                                                              
+         elif isinstance(c, Content):  # Already a Content instance                                                                                                           
+             result.append(c)                                                                                                                                                 
+         elif isinstance(c, str):  # Plain text handling                                                                                                                      
+             result.append(Text(text=c))                                                                                                                                      
+         else:                                                                                                                                                                
+             # Handle unexpected content formats if necessary                                                                                                                 
+             raise ValueError(f"Unsupported content type: {type(c)}")                                                                                                         
+     return result       
```

Comment:
Before updating `content_converter` I would get the following:
```
 "Traceback (most recent call last):
  File \"/Users/aarong/Development/goose/packages/exchange/src/exchange/exchange.py\", line 149, in call_function
    output = json.dumps(tool.function(**tool_use.parameters))
 File \"/Users/aarong/Development/goose/src/goose/toolkit/vision.py\", line 22, in describe_image
    user_message = Message(role=\"user\", content=[f\"{instructions}: \", image])
  File \"<attrs generated init exchange.message.Message>\", line 13, in __init__

    _setattr('content', __attr_converter_content(content))
  File \"/Users/aarong/Development/goose/packages/exchange/src/exchange/message.py\", line 29, in content_converter\n    return [(CONTENT_TYPES[c.pop(\"type\")](**c) if c.__class__ not in CONTENT_TYPES.values() else c) for c in contents]

  File \"/Users/aarong/Development/goose/packages/exchange/src/exchange/message.py\", line 29, in <listcomp>\n    return [(CONTENT_TYPES[c.pop(\"type\")](**c) if c.__class__ not in CONTENT_TYPES.values() else c) for c in contents]\nAttributeError: 'str' object has no attribute 'pop'

'str' object has no attribute 'pop'", "is_error": true, "type": "ToolResult"}]
```
![image](https://github.com/user-attachments/assets/3151bdc0-092b-4719-8111-91548a394ff4)